ast/Helpers.h

```diff
@@ -175,8 +175,7 @@ class MK {
     }
 
     static ExpressionPtr Splat(core::LocOffsets loc, ExpressionPtr arg) {
-        auto to_a = Send0(loc, std::move(arg), core::Names::toA());
-        return Send1(loc, Constant(loc, core::Symbols::Magic()), core::Names::splat(), std::move(to_a));
+        return Send1(loc, Constant(loc, core::Symbols::Magic()), core::Names::splat(), std::move(arg));
```

Comment:
With this change, `x = *1` would go from `x = Magic::splat(1.to_a)` to `x = Magic::splat(1)`. Is that expected?

Reply:
Yes; `1.to_a` throws `NoMethodError`, while `*1` returns `[1]`.  We want the latter, not the former.